include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl.h

```diff
@@ -577,6 +580,103 @@ struct __parallel_transform_scan_static_single_group_submitter<_Inclusive, _Elem
     }
 };
 
+template <typename _Size, ::std::uint16_t _ElemsPerItem, ::std::uint16_t _WGSize, bool _IsFullGroup,
+          typename _KernelName>
+struct __parallel_copy_if_static_single_group_submitter;
+
+template <typename _Size, ::std::uint16_t _ElemsPerItem, ::std::uint16_t _WGSize, bool _IsFullGroup,
+          typename... _ScanKernelName>
+struct __parallel_copy_if_static_single_group_submitter<_Size, _ElemsPerItem, _WGSize, _IsFullGroup,
+                                                        __internal::__optional_kernel_name<_ScanKernelName...>>
+{
+    template <typename _Policy, typename _InRng, typename _OutRng, typename _InitType, typename _BinaryOperation,
+              typename _UnaryOp>
+    auto
+    operator()(const _Policy& __policy, _InRng&& __in_rng, _OutRng&& __out_rng, ::std::size_t __n, _InitType __init,
+               _BinaryOperation __bin_op, _UnaryOp __unary_op)
+    {
+        using _ValueType = ::std::uint16_t;
+
+        // This type is used as a workaround for when an internal tuple is assigned to ::std::tuple, such as
+        // with zip_iterator
+        using __tuple_type = typename ::oneapi::dpl::__internal::__get_tuple_type<
+            typename ::std::decay_t<decltype(__in_rng[0])>, typename ::std::decay_t<decltype(__out_rng[0])>>::__type;
```

Comment:
Despite the comment, I'm confused as to why this and the `static_cast` later on are necessary in this pattern.  

I think I understand `__get_tuple_type<>` generally, its so that we can convert our internal tuple to a `std::tuple`.  It seems to be used for the other users of `__parallel_scan_copy`, but I'm wondering if that is because of their internal use of zip iterator.  Is it still necessary here?  

If so, should we be using this elsewhere?

Reply:
This issue is exposed in the `zip_iterator.pass` test, which calls `copy_if` with a `zip_iterator` input iterator and a `transform_iterator` output iterator. Without the cast, we are met with the following error:
```
[...]/include/oneapi/dpl/pstl/tuple_impl.h:261:31: error: reference to type 'const std::_Swallow_assign' could not bind to an lvalue of type 'const int'
    __value_holder(_Up&& t) : value(::std::forward<_Up>(t))
```
This is the same workaround used in the `__copy_by_mask` functor for the original `copy_if` algorithm. See the comment at https://github.com/oneapi-src/oneDPL/blob/6bb45531b7bd6908a16897ebd8fedbb18d365430/include/oneapi/dpl/pstl/hetero/dpcpp/unseq_backend_sycl.h#L517-L535

Reply:
Thanks for the additional information.  I think it is fine here in the context of this PR.

I'll make an issue to discuss this `__get_tuple_type<>` workaround in general.  I think it is there to enable a usage pattern which should be discouraged (and can be rewritten to not require the workaround). Also, there are other algorithm(s?) for which a similar usage pattern will cause the same compilation errors that don't currently use this workaround.